test/test_numpy_interop.py

```diff
@@ -153,6 +153,29 @@ def test_to_numpy_bool(self, device) -> None:
         self.assertEqual(y.dtype, np.bool_)
         self.assertEqual(x[0], y[0])
 
+    def test_to_numpy_force_argument(self, device) -> None:
+        for force in [False, True]:
+            for requires_grad in [False, True]:
+                for sparse in [False, True]:
+                    for conj in [False, True]:
+                        data = [[1 + 2j, -2 + 3j], [-1 - 2j, 3 - 2j]]
+                        x = torch.tensor(data, requires_grad=requires_grad, device=device)
+                        y = x
+                        if sparse:
+                            if requires_grad:
+                                continue
+                            x = x.to_sparse()
+                        if conj:
+                            x = x.conj()
+                            y = x.resolve_conj()
+                        expect_error = requires_grad or sparse or conj or not device == 'cpu'
+                        if not force and expect_error:
+                            self.assertRaises((RuntimeError, TypeError), lambda: x.numpy())
+                        elif force and sparse:
+                            self.assertRaises(TypeError, lambda: x.numpy())
```

Comment:
Shouldn't the lambda here be `x.numpy(force=True)`?

Reply:
The one I prefer by the way is:
```
with self.assertRaisesRegex(TypeError, "part of the error message"):
    x.numpy(force=True)
```

This makes sure that the error that is raised is the one you expect and not another unrelated TypeError.

Reply:
in last 2 commits i have unified error messages you can get in tensor.numpy(). CI failed, because some test used `self.assertRaisesRegex()`. Is my change breaking BC ? Should i return to the previous eroor messages ?

Reply:
It is ok to update the error message. We don't consider this BC in general.
You can just update the tests to make them pass again.